caikit_computer_vision/modules/segmentation/image_processing.py

```diff
@@ -0,0 +1,281 @@
+        multipolygon = spg.MultiPolygon(
+            [
+                spg.Polygon(list(zip(poly[0::2], poly[1::2])))
+                for poly in ann["segmentation"]
+            ]
+        )
+        ann["area"] = int(multipolygon.area)
+
+    # extract seg array from list of segs and cast to list
+    for idx, ann in enumerate(inf_coco.dataset["annotations"]):
+        seg = [i.tolist() for i in ann["segmentation"]]
+        inf_coco.dataset["annotations"][idx]["segmentation"] = seg
+
+    # save annotations
+    if save_coco:
+        if not save_coco.endswith("/"):
+            save_coco = save_coco + "/"
+        with open(save_coco + "detections_poly.json", "w") as json_file:
+            json.dump(inf_coco.dataset, json_file, sort_keys=True, indent=4)
+
+    return inf_coco
+
+
+def convert_seg_poly(inf_coco: Type[COCO]) -> Type[COCO]:
+    """converts segmentation from detectron2 format to polygons.
+
+    Args:
+      inf_coco : coco object with annotations containing detectron2 formatted segmentations
+
+    Returns:
+      inf_coco : coco dict with annotations containing coco formatted polygons
+    """
+
+    def mask_to_polygons(mask):
+        """from Detectron2 code"""
+        # cv2.RETR_CCOMP flag retrieves all the contours and arranges them to a 2-level
+        # hierarchy. External contours (boundary) of the object are placed in hierarchy-1.
+        # Internal contours (holes) are placed in hierarchy-2.
+        # cv2.CHAIN_APPROX_NONE flag gets vertices of polygons from contours.
+        mask = np.ascontiguousarray(
+            mask
+        )  # some versions of cv2 does not support incontiguous arr
+        res = cv2.findContours(
+            mask.astype("uint8"), cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE
+        )
+        hierarchy = res[-1]
+        if hierarchy is None:  # empty mask
+            return [], False
+        has_holes = (hierarchy.reshape(-1, 4)[:, 3] >= 0).sum() > 0
+        res = res[-2]
+        res = [x.flatten() for x in res]
+        # These coordinates from OpenCV are integers in range [0, W-1 or H-1].
+        # We add 0.5 to turn them into real-value coordinate space. A better solution
+        # would be to first +0.5 and then dilate the returned polygon by 0.5.
+        res = [x + 0.5 for x in res if len(x) >= 6]
+        return res, has_holes
+
+    for ann in tqdm(inf_coco.dataset["annotations"]):
+        img = inf_coco.loadImgs([ann["image_id"]])
+        height, width = img[0]["height"], img[0]["width"]
+        segm = ann["segmentation"]
+
+        if isinstance(segm["counts"], list):
+            seg = maskUtils.frPyObjects(segm, height, width)
+            mask = inf_coco.annToMask(seg)
+        else:
+            mask = inf_coco.annToMask(ann)
+
+        mask = np.array(mask, dtype=np.uint8) * 255
+        # we arrive at some error without two lines below, after these operations mask remains unchanged
```

Comment:
This is a bit strange, what is the error that comes without the two lines mentioned here? 

Maybe the binary mask in line 274 is a third order tensor with axis dim[2] == 1, which gets squeezed by the conversion maybe...? I'm not familiar with annToMask, but it seems like lines 276-277 shouldn't leave the mask unchanged from 274

Reply:
Okay - I think it might be worth investigating removing it in the future to avoid unnecessary conversions, but for now we can leave it